### PR TITLE
Add file system(EFS) functionality.

### PR DESCRIPTION
**Pull request by welderpb**
## what
Add file system access for Lambda functions.
https://docs.aws.amazon.com/lambda/latest/dg/configuration-filesystem.html

### CODE REVIEW

File: iam-role.tf [hunk 1]
```diff
@@ -47,6 +47,35 @@ resource "aws_iam_role_policy_attachment" "xray" {
   role       = aws_iam_role.this[0].name
 }
 
+# Allow Lambda to access specific EFS"
+data "aws_iam_policy_document" "file_system" {
```
Comment (Gowiem) [inline iam-role.tf:51]: Can you provide a `sid` here to make this more descriptive?

Comment (welderpb) [reply iam-role.tf:51]: Done.

**Comment by max-lobur** [comment]
Please rebase on main, or just delete the `.github/workflows/validate-codeowners.yml` - it was removed from main.

**Comment by Gowiem** [comment]
@welderpb sorry for the delay on this -- mind revisit this + fixing conflicts and we can work to get it merged? Also, please run the following commands, commit, and push the changes:

```
make init
make github/init
make readme
```

Thanks!

**Pull request closed without merging**